sdk/eventhub/event-hubs/review/event-hubs.api.md

```diff
@@ -229,11 +247,37 @@ export type OnError = (error: MessagingError | Error) => void;
 // @public
 export type OnMessage = (eventData: ReceivedEventData) => void;
 
+// @public (undocumented)
+export type OnReceivedEvents = (receivedEvents: ReceivedEventData[], context: PartitionContext) => Promise<void>;
+
+// @public
+export interface OptionalEventHandlers {
+    // (undocumented)
+    onClose?: (reason: CloseReason, context: PartitionContext) => Promise<void>;
+    // (undocumented)
+    onError?: (error: Error, context: PartitionContext) => Promise<void>;
+    // (undocumented)
+    onInitialize?: (context: PartitionContext) => Promise<void>;
+}
+
 // @public
 export interface ParentSpanOptions {
     parentSpan?: Span | SpanContext;
 }
 
+// @public
+export interface PartitionContext {
```

Comment:
One thing missing with all these changes is a way to call updateCheckpoint given an ReceivedEventData.

Currently this method exists on the PartitionProcessor. I think we discussed with the python team about putting in on the PartitionContext (it actually used to live here once upon a time). Could you add it to the PartitionContext again?

For now, only the load-balanced subscribe call can call updateCheckpoint since you need a PartitionManager to perform checkpointing. It could be useful to throw an error telling the user to provide a PartitionManager to enable checkpointing if they try to do it without passing in a PartitionManager.

After we split ownership/checkpointing we can revisit supporting checkpointing for every subscribe overload, but we at least need it for the load-balanced subscribe.

Reply:
Got it. It's good - there's a checkpointer that's passed to each handler.